[Keycloak Operator CI] - Test OLM Installation

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

`error parsing HTTP 499 response body: invalid character '<' looking for beginning of value: "<!DOCTYPE html>\n<html>\n  <head>\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n    <meta charset=\"utf-8\">\n    <title>Application Error</title>\n    <style media=\"screen\">\n      html,body,iframe {\n        margin: 0;\n        padding: 0;\n      }\n\n      html,body {\n        height: 100%;\n        overflow: hidden;\n      }\n\n      iframe {\n        width: 100%;\n        height: 100%;\n        border: 0;\n      }\n    </style>\n  </head>\n  <body>\n    <iframe src=\"[https://www.herokucdn.com/error-pages/application-error.html\](https://www.herokucdn.com/error-pages/application-error.html/)"></iframe>\n  </body>\n</html>"`
Error: Process completed with exit code 1.

https://github.com/keycloak/keycloak/actions/runs/15336283564/job/43154257333#step:9:5913
https://github.com/keycloak/keycloak/actions/runs/15324971197/job/43135161773#step:9:769
https://github.com/keycloak/keycloak/actions/runs/15322156793/job/43108326992#step:9:762
https://github.com/keycloak/keycloak/actions/runs/15307248702/job/43063268179#step:9:773
https://github.com/keycloak/keycloak/actions/runs/15306958721/job/43062250593#step:7:41

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/.github/workflows/operator-ci.yml
+++ b/.github/workflows/operator-ci.yml
@@ -143,7 +143,7 @@ jobs:
           kubernetes version: ${{ env.KUBERNETES_VERSION }}
           github token: ${{ secrets.GITHUB_TOKEN }}
           driver: docker
-          start args: --memory=${{ env.MINIKUBE_MEMORY }}
+          start args: --memory=${{ env.MINIKUBE_MEMORY }} --addons=registry --insecure-registry=192.168.49.2:5000
 
       - name: Install OPM
         uses: redhat-actions/openshift-tools-installer@144527c7d98999f2652264c048c7a9bd103f8a82 # v1.13.1
@@ -169,7 +169,7 @@ jobs:
         working-directory: operator
         run: |
           eval $(minikube -p minikube docker-env)
-          ./scripts/olm-testing.sh ${GITHUB_SHA::6}
+          REGISTRY=192.168.49.2:5000 ./scripts/olm-testing.sh ${GITHUB_SHA::6}
 
       - name: Deploy an example Keycloak and wait for it to be ready
         working-directory: operator

--- a/operator/scripts/create-olm-test-catalog.sh
+++ b/operator/scripts/create-olm-test-catalog.sh
@@ -19,7 +19,7 @@ mkdir -p $SCRIPT_DIR/../olm/catalog/test-catalog
     --output yaml > test-catalog/operator.yaml
 
   opm render $BUNDLE_IMAGE:$VERSION \
-    --output=yaml >> test-catalog/operator.yaml
+    --output=yaml --skip-tls >> test-catalog/operator.yaml
 
   cat << EOF >> test-catalog/operator.yaml
 ---

--- a/operator/scripts/olm-testing.sh
+++ b/operator/scripts/olm-testing.sh
@@ -10,14 +10,15 @@ TARGET_NAMESPACES=${3-$INSTALL_NAMESPACE}
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 # This version translates to one day for ttl.sh
+REGISTRY=${REGISTRY:-ttl.sh}
 VERSION="86400000.0.0"
 
 # Build Keycloak Docker image (the keycloak tar.gz should already be in the container folder)
 (
   cd $SCRIPT_DIR/../../quarkus/container
   
-  docker build --build-arg KEYCLOAK_DIST=$(ls keycloak-*.tar.gz) . -t "ttl.sh/${UUID}keycloak:${VERSION}"
-  docker push "ttl.sh/${UUID}keycloak:${VERSION}"
+  docker build --build-arg KEYCLOAK_DIST=$(ls keycloak-*.tar.gz) . -t "$REGISTRY/${UUID}keycloak:${VERSION}"
+  docker push "$REGISTRY/${UUID}keycloak:${VERSION}"
 )
 
 # Build the operator Docker image
@@ -25,13 +26,13 @@ VERSION="86400000.0.0"
   cd $SCRIPT_DIR/../../
   ./mvnw clean package -Poperator -pl :keycloak-operator -am \
     -Dquarkus.container-image.build=true \
-    -Dquarkus.container-image.image="ttl.sh/${UUID}keycloak-operator:${VERSION}" \
-    -Dkc.operator.keycloak.image="ttl.sh/${UUID}keycloak:${VERSION}" \
+    -Dquarkus.container-image.image="$REGISTRY/${UUID}keycloak-operator:${VERSION}" \
+    -Dkc.operator.keycloak.image="$REGISTRY/${UUID}keycloak:${VERSION}" \
     -DskipTests
   # JIB patching on images doesn't work reliably with ttl.sh
-  docker push "ttl.sh/${UUID}keycloak-operator:${VERSION}"
+  docker push "$REGISTRY/${UUID}keycloak-operator:${VERSION}"
 )
 
-$SCRIPT_DIR/prepare-olm-test.sh ttl.sh ${VERSION} NONE ${UUID} $TARGET_NAMESPACES
+$SCRIPT_DIR/prepare-olm-test.sh $REGISTRY ${VERSION} NONE ${UUID} $TARGET_NAMESPACES
 
 $SCRIPT_DIR/install-keycloak-operator.sh $INSTALL_NAMESPACE

